Lib/test/test_urllib.py

```diff
@@ -1445,7 +1445,7 @@ def test_thishost(self):
         self.assertIsInstance(urllib.request.thishost(), tuple)
 
 
-class URLopener_Tests(unittest.TestCase):
+class URLopener_Tests(unittest.TestCase, FakeHTTPMixin):
```

Comment:
The mixin class should be applied first.

Reply:
All the tests in `test_urllib` use `Mixin` after `TestCase`. Grepping tells me both variants are used across test suite. Is there any difference I am missing since I am using the mixin only for `fakehttp`?

Reply:
Yes, there are a dozen of incorrect usages in the test suite (mainly in Lib/test/test_ipaddress.py, Lib/test/test_marshal.py, Lib/test/test_posix.py, Lib/test/test_runpy.py, Lib/test/test_statistics.py, Lib/test/test_with.py) but that doesn't justify adding non-idiomatic code to the test suite just because we are only using `some_unique_helper` from the mixin class.

Contributors often use existing tests as examples when writing new tests for their PRs, so it would be nice to write idiomatic tests as much as possible.

Reply:
Changed as suggested, thanks.